### ISSUE TITLE
Bug: no-restricted-imports negate pattern doesn't work

### ISSUE THREAD

**Issue by Shakeskeyboarde** [eslint/eslint#15057]
### Environment

Node version: 16.1.0
npm version: 7.17.0
Local ESLint version: 7.32.0
Global ESLint version: none
Operating System: OSX

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
  ],
  plugins: ['@typescript-eslint'],
  parser: '@typescript-eslint/parser',
  rules: {
    'no-restricted-imports': ['error', { patterns: ['../*', '!*/support/*'] }],
  },
};
```

</details>

<details>
<summary>Test Case</summary>

```ts
import { foo } from '../foo'; // Should error
import { bar } from '../support/bar'; // Should NOT error because of the negation pattern.
```

</details>

### What did you expect to happen?

I'm trying to disallow relative parent imports, _except_ when the path contains the `/support/` directory. So, the `../*` pattern should cause any import starting with a parent relative prefix to be an error, and the `!*/support/*` pattern should cause the rule to ignore any paths which contain the `/support/` directory (even if they start with `../`).

### What actually happened?

All parent imports, including those that contain `/support/`, are reported as errors.

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @Shakeskeyboarde, thanks for the issue! 

> ```js
> 'no-restricted-imports': ['error', { patterns: ['../*', '!*/support/*'] }],
> ```

> ```ts
> import { foo } from '../foo'; // Should error
> import { bar } from '../support/bar'; // Should NOT error because of the negation pattern.
> ```

This works as intended. Per the documentation for this rule, patterns are gitignore-style patterns, and per the [gitignore specification](https://git-scm.com/docs/gitignore) it is not possible to re-include a file if a parent directory of that file is excluded.

`../*` ignores the content of `../`, including `../support`.

`!*/support/*` unignores the content of the `../support/` directory, but the directory itself is still ignored.

If you want to unignore the directory, remove `*` from the end:

```
'no-restricted-imports': ['error', { patterns: ['../*', '!*/support/'] }]
```

```js
import { foo } from '../foo'; // error
import { bar } from '../support/bar'; // no error
```
